modules/carto/src/config.ts

```diff
@@ -4,17 +4,33 @@ export const defaultClassicCredentials = {
   apiKey: 'default_public',
   region: 'us',
   username: 'public'
-};
+} as const;
 
 export const defaultCloudNativeCredentials = {
   apiBaseUrl: 'https://gcp-us-east1.api.carto.com'
-};
+} as const;
+
+export interface ClassicCredentials {
+  apiVersion: 'v1' | 'v2';
+  apiKey: string;
+  region: string;
+  username: string;
+  mapsUrl?: string;
+}
+
+export interface CloudNativeCredentials {
+  apiVersion: 'v3';
+  apiBaseUrl: string;
+  accessToken?: string;
+  mapsUrl?: string;
+}
 
-let credentials = {};
+type Credentials = ClassicCredentials | CloudNativeCredentials;
+let credentials: Credentials = {} as Credentials;
 
 setDefaultCredentials({});
 
-export function setDefaultCredentials({apiVersion, ...rest}) {
+export function setDefaultCredentials({apiVersion, ...rest}: Partial<Credentials>) {
```

Comment:
Does this have to be Partial? I would instead mark the properties that are optional as optional (that is, everything except `apiVersion`).

Reply:
In this signature everything is optional, including `apiVersion`. So yes, this is partial.

With those typings, i tried to convey fact that `setDefaultCredentials` can accept almost everything (partial), but `getDefaultCredentials` will return actual "Credentials" for which some of them are actually mandatory and/or filled using defaults.

So, it depends.

With this typings, instantiating `Credentials` is close-to typesafe.

Reply:
Ideally it shouldn't accept everything. E.g. this mix of parameters is invalid:

```javascript
{
  apiVersion: 'v1',
  accessToken: 'XXXX'
}
```

Would `Partial<ClassicCredentials> | Partial<CloudNativeCredentials>` work?